routes/main.js

```diff
@@ -0,0 +1,218 @@
+				return res.status(404).send('Product not found');
+			}
+			res.status(500).send('Server Error');
+		});
+});
+
+router.get('/products/:product/reviews', async (req, res) => {
+	const productId = req.params.product;
+	const page = parseInt(req.query.page) || 1;
+	const reviewsPerPage = 4;
+	const skip = (page - 1) * reviewsPerPage;
+
+	try {
+		const product = await Product.findById(productId).populate({
+			path: 'reviews',
+			options: {
+				skip: skip,
+				limit: reviewsPerPage,
+			},
+		});
+
+		if (!product) {
+			return res.status(404).send('Product not found');
+		}
+		const totalReviews = await Review.countDocuments({
+			product: productId,
+		});
+
+		res.status(200).json({
+			reviews: product.reviews,
+			currentPage: page,
+			totalPages: Math.ceil(totalReviews / reviewsPerPage),
+		});
+	} catch (err) {
+		if (err.name === 'CastError') {
+			return res.status(404).send('Product not found');
+		}
+		res.status(500).send('Server Error');
+	}
+});
+
+router.post('/products', async (req, res) => {
+	const { category, name, price, image } = req.body;
+
+	if (!category || !name || !price || !image) {
+		return res.status(400).send('All fields are required');
+	}
+
+	try {
+		const newProduct = new Product({
+			category,
+			name,
+			price,
+			image,
+		});
+
+		const savedProduct = await newProduct.save();
+
+		res.status(200).send(savedProduct);
+	} catch (err) {
+		res.status(500).send('Server Error');
+	}
+});
+
+router.post('/products/:product/reviews', async (req, res) => {
+	const productId = req.params.product;
+	const { username, text } = req.body;
+
+	if (!username || !text) {
+		return res.status(400).send('Username and text are required');
+	}
+
+	try {
+		const product = await Product.findById(productId);
+		if (!product) {
+			return res.status(404).send('Product not found');
+		}
+
+		const review = new Review({
+			username,
+			text,
+			product: productId,
+		});
+
+		const savedReview = await review.save();
+
+		product.reviews.push(savedReview._id);
+		await product.save();
+
+		res.status(200).send(savedReview);
+	} catch (err) {
+		if (err.name === 'CastError') {
+			return res.status(404).send('Product not found');
+		}
+		res.status(500).send('Server Error');
+	}
+});
+
+router.delete('/products/:product', async (req, res) => {
+	const productId = req.params.product;
+
+	try {
+		const deletedProduct = await Product.findByIdAndDelete(productId);
+
+		if (!deletedProduct) {
+			return res.status(404).send('Product not found');
+		}
+
+		res.status(200).send('Product deleted successfully');
+	} catch (err) {
+		if (err.name === 'CastError') {
+			return res.status(404).send('Product not found');
+		}
+		res.status(500).send('Server error');
+	}
+});
+
+router.delete('/reviews/:review', async (req, res) => {
+	const reviewId = req.params.review;
+
+	try {
+		const deletedReview = await Review.findByIdAndDelete(reviewId);
+
+		if (deletedReview) {
+			return res.status(404).send('Review not found');
```

Comment:
this is a bug, if you find a deleted review, you are sending an error instead of deleting it
```suggestion
		if (!deletedReview) {
			return res.status(404).send('Review not found');
```